### PR TITLE
Documentation updated for Lots, Events and Credits

### PR DESCRIPTION
**Pull request by edy2310**
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

### Changelog
Please ensure to add changelog with this PR by answering the following questions.
  1. What's the purpose of the update?    
      - [ ] new service onboarding 
      - [ ] new API version 
      - [x] update existing version for new feature 
      - [ ] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify 
  2. When you are targeting to deploy new service/feature to public regions? Please provide date, or month to public if date is not available yet.
  3. When you expect to publish swagger? Please provide date, or month to public if date is not available yet.
  4. If it's an update to existing version,  please select SDKs of specific language and CLIs that require refresh after swagger is published.
      - [ ] SDK of .NET (need service team to ensure code readiness)
      - [ ] SDK of Python
      - [ ] SDK of Java
      - [ ] SDK of Js
      - [ ] SDK of Go
      - [ ] PowerShell
      - [ ] CLI
      - [ ] Terraform
      - [x] No, no need to refresh for updates in this PR

### Contribution checklist:
- [ ] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of "no breaking changes"
- [x] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [x] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist
- [ ] Ensure to check this box if one of the following scenarios meet updates in the PR, so that label “WaitForARMFeedback” will be added automatically to involve ARM API Review. Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs, all “removals” and “adding a new property” no more require ARM API review.
  - Adding new API(s)
  - Adding a new API version
  -  [ ] Ensure to copy the existing version into new directory structure for first commit (including refactoring) and then push new changes including version updates in separate commits. This is required to review the changes efficiently.
  - Adding a new service

- [ ] Please ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If there are following updates in the PR, ensure to request an approval from Breaking Change Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 

- [ ] Removing API(s) in stable version
- [ ] Removing properties in stable version
- [ ] Removing API version(s) in stable version
- [ ] Updating API in stable or public preview version with Breaking Change Validation errors
- [ ] Updating API(s) in public preview over 1 year (refer to [Retirement of Previews](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37683/Retirement-of-Previews))

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Addition details on the process and office hours are on the [Breaking change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

File: specification/consumption/resource-manager/Microsoft.Consumption/stable/2021-05-01/consumption.json [hunk 1]
```diff
@@ -5113,6 +5186,16 @@
...
+          "readOnly": true
+        },
+        "billingCurrency": {
+          "description": "The billing currency.",
+          "type": "string",
+          "readOnly": true
+        },
         "balanceSummary": {
           "description": "Summary of balances associated with this credit summary.",
           "readOnly": true,
```
Comment (karthikku-2020) [inline specification/consumption/resource-manager/Microsoft.Consumption/stable/2021-05-01/consumption.json:5201]: CreditBalanceSummary definition is missing "EstimatedBalanceInBillingCurrency"

File: specification/consumption/resource-manager/Microsoft.Consumption/stable/2021-05-01/consumption.json [hunk 2]
```diff
@@ -5166,6 +5254,38 @@
         }
       }
     },
+    "AmountWithExchangeRate": {
+      "description": "The amount with exchange rate.",
+      "type": "object",
+      "properties": {
+        "exchangeRate": {
+          "description": "The exchange rate.",
+          "type": "decimal",
```
Comment (karthikku-2020) [inline specification/consumption/resource-manager/Microsoft.Consumption/stable/2021-05-01/consumption.json:5263]: "type": "number",  "format": "decimal",

File: specification/consumption/resource-manager/Microsoft.Consumption/stable/2021-05-01/consumption.json [hunk 3]
```diff
@@ -5166,6 +5254,38 @@
...
+          "readOnly": true
+        },
+        "exchangeRateMonth": {
+          "description": "The exchange rate month.",
+          "type": "int",
```
Comment (karthikku-2020) [inline specification/consumption/resource-manager/Microsoft.Consumption/stable/2021-05-01/consumption.json:5268]: "type": "integer", "format": "int32",

File: specification/consumption/resource-manager/Microsoft.Consumption/stable/2021-05-01/consumption.json [hunk 4]
```diff
@@ -5166,6 +5254,38 @@
         }
       }
     },
+    "AmountWithExchangeRate": {
+      "description": "The amount with exchange rate.",
+      "type": "object",
```
Comment (karthikku-2020) [inline specification/consumption/resource-manager/Microsoft.Consumption/stable/2021-05-01/consumption.json:5259]: type would not be needed here.

File: specification/consumption/resource-manager/Microsoft.Consumption/stable/2021-05-01/consumption.json [hunk 5]
```diff
@@ -5089,6 +5154,14 @@
...
           "description": "The status of the lot.",
           "type": "string",
+          "enum": [
+            "None",
+            "Active",
+            "Inactive",
+            "Expired",
+            "Complete",
+            "Canceled"
+          ],
```
Comment (karthikku-2020) [inline specification/consumption/resource-manager/Microsoft.Consumption/stable/2021-05-01/consumption.json:5164]: The following needs to be added

"x-ms-enum": {
            "name": "Status",
            "modelAsString": true
          }

File: specification/consumption/resource-manager/Microsoft.Consumption/stable/2021-05-01/examples/LotsListByBillingProfile.json [hunk 6]
```diff
@@ -13,37 +13,69 @@
...
+              "reseller": {
+                "resellerId": "/providers/Microsoft.Billing/billingAccounts/1234:5678/billingProfiles/2468/providers/Microsoft.Consumption/reseller/reseller2",
+                "resellerDescription": 2
+              },
               "source": "PurchasedCredit",
               "startDate": "2021-05-01T00:00:00Z",
               "expirationDate": "2019-12-31T00:00:00Z",
-              "poNumber": "31224"
+              "poNumber": "31224",
+              "purchaseDate": "2019-09-01T00:00:00Z",
```
Comment (karthikku-2020) [inline specification/consumption/resource-manager/Microsoft.Consumption/stable/2021-05-01/examples/LotsListByBillingProfile.json:77]: purchaseDate and status won't be available at the billingProfile scope.

File: specification/consumption/resource-manager/Microsoft.Consumption/stable/2021-05-01/examples/LotsListByBillingAccountWithFilters.json [hunk 7]
```diff
@@ -13,17 +13,32 @@
             "name": "lot1",
             "type": "Microsoft.Consumption/lots",
             "properties": {
+              "creditCurrency": "USD",
```
Comment (karthikku-2020) [inline specification/consumption/resource-manager/Microsoft.Consumption/stable/2021-05-01/examples/LotsListByBillingAccountWithFilters.json:16]: Please revert all the changes on this file.

File: specification/consumption/resource-manager/Microsoft.Consumption/stable/2021-05-01/examples/LotsListByBillingAccount.json [hunk 8]
```diff
@@ -12,17 +12,32 @@
             "name": "lot1",
             "type": "Microsoft.Consumption/lots",
             "properties": {
+              "creditCurrency": "USD",
```
Comment (karthikku-2020) [inline specification/consumption/resource-manager/Microsoft.Consumption/stable/2021-05-01/examples/LotsListByBillingAccount.json:15]: Please revert all the changes in this file.

File: specification/consumption/resource-manager/Microsoft.Consumption/stable/2021-05-01/consumption.json (hunk 1 above)
Comment (edy2310) [reply specification/consumption/resource-manager/Microsoft.Consumption/stable/2021-05-01/consumption.json:5205]: Added.

File: specification/consumption/resource-manager/Microsoft.Consumption/stable/2021-05-01/consumption.json (hunk 2 above)
Comment (edy2310) [reply specification/consumption/resource-manager/Microsoft.Consumption/stable/2021-05-01/consumption.json]: Done.

File: specification/consumption/resource-manager/Microsoft.Consumption/stable/2021-05-01/consumption.json (hunk 4 above)
Comment (edy2310) [reply specification/consumption/resource-manager/Microsoft.Consumption/stable/2021-05-01/consumption.json]: Removed.

File: specification/consumption/resource-manager/Microsoft.Consumption/stable/2021-05-01/consumption.json (hunk 5 above)
Comment (edy2310) [reply specification/consumption/resource-manager/Microsoft.Consumption/stable/2021-05-01/consumption.json:5164]: Added.

File: specification/consumption/resource-manager/Microsoft.Consumption/stable/2021-05-01/examples/LotsListByBillingProfile.json (hunk 6 above)
Comment (edy2310) [reply specification/consumption/resource-manager/Microsoft.Consumption/stable/2021-05-01/examples/LotsListByBillingProfile.json]: Removed.

File: specification/consumption/resource-manager/Microsoft.Consumption/stable/2021-05-01/examples/LotsListByBillingAccountWithFilters.json (hunk 7 above)
Comment (edy2310) [reply specification/consumption/resource-manager/Microsoft.Consumption/stable/2021-05-01/examples/LotsListByBillingAccountWithFilters.json]: Reverted.

File: specification/consumption/resource-manager/Microsoft.Consumption/stable/2021-05-01/examples/LotsListByBillingAccount.json (hunk 8 above)
Comment (edy2310) [reply specification/consumption/resource-manager/Microsoft.Consumption/stable/2021-05-01/examples/LotsListByBillingAccount.json]: Reverted.

**Comment by edy2310** [comment]
@karthikku-2020 I resolved all your comments and made the changes to the Events examples. Please let me know if I made the correct changes.

**Comment by ChenTanyi** [comment]
@edy2310 Have you released .NET SDK? If the API you change has released .NET SDK, please move all properties you added to the end which is required by .NET SDK.

File: specification/consumption/resource-manager/Microsoft.Consumption/stable/2021-05-01/examples/EventsGetByBillingAccount.json [hunk 9]
```diff
@@ -34,6 +42,14 @@
             "name": "eventId2",
             "type": "Microsoft.Consumption/events",
             "properties": {
+              "newCredit": {
+                "currency": "USD",
+                "value": 500
+              },
+              "creditExpired": {
```
Comment (karthikku-2020) [inline specification/consumption/resource-manager/Microsoft.Consumption/stable/2021-05-01/examples/EventsGetByBillingAccount.json:49]: Same as above. This should be canceledCredit.

File: specification/consumption/resource-manager/Microsoft.Consumption/stable/2021-05-01/examples/EventsGetByBillingAccountWithFilters.json [hunk 10]
```diff
@@ -13,6 +13,14 @@
             "name": "eventId1",
             "type": "Microsoft.Consumption/events",
             "properties": {
+              "newCredit": {
+                "currency": "USD",
+                "value": 500
+              },
+              "creditExpired": {
```
Comment (karthikku-2020) [inline specification/consumption/resource-manager/Microsoft.Consumption/stable/2021-05-01/examples/EventsGetByBillingAccountWithFilters.json:20]: This should be canceledCredit.

**Review by karthikku-2020 — APPROVED** [review]

**Comment by edy2310** [comment]
> Hi @edy2310, one or multiple breaking change(s) is detected in your PR. Please check out the breaking change(s), and provide business justification in the PR comment and @ PR assignee why you must have these change(s), and how external customer impact can be mitigated. Please ensure to follow [breaking change policy](https://aka.ms/AzBreakingChangesPolicy) to request breaking change review and approval before proceeding swagger PR review.
> **Action**: To initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Addition details on the process and office hours are on the [Breaking change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).
> If you want to know the production traffic statistic, please see [ARM Traffic statistic](https://portal.azure-devex-tools.com/api/pr/query-traffic?pr=Azure/azure-rest-api-specs/14314&days=5).
> If you think it is false positive breaking change, please provide the reasons in the PR comment, report to Swagger Tooling Team via https://aka.ms/swaggerfeedback.

After reviewing the messages from BreakingChanges, I consider these to be a false positive as the updated documentation will not be used by external clients or internal partners.

**Comment by ChenTanyi** [comment]
@edy2310 You could tell the braking change review board about it. As it breaks in swagger file.

**Review by JeffreyRichter — APPROVED** [review]

**Comment by ChenTanyi** [comment]
@edy2310 Please tell me when it ready to merge.

**Pull request merged**